Sources/MongoSwift/MongoError.swift

```diff
@@ -260,33 +269,26 @@ internal func extractMongoError(error bsonError: bson_error_t, reply: Document?
     }
 }
 
-/// Internal function used to get an appropriate error from a libmongoc error and/or a server reply to a
-/// `BulkWriteOperation`. If a `ServerError.bulkWriteError` is extracted and a partial result is provided, an updated
-/// result with the failed results filtered out will be returned as part of the error.
-internal func extractMongoError(bulkOp: BulkWriteOperation,
-                                error: bson_error_t,
-                                reply: Document? = nil,
-                                partialResult: BulkWriteResult? = nil) -> Error {
-    // if the reply is nil or writeErrors or writeConcernErrors aren't present, then this is likely a commandError.
-    guard let serverReply = reply,
-          !(serverReply["writeErrors"] as? [BSONValue] ?? []).isEmpty ||
-          !(serverReply["writeConcernErrors"] as? [BSONValue] ?? []).isEmpty else {
-        return parseMongocError(error, reply: reply)
-    }
-
+/// Internal function used to get a `ServerError.bulkWriteError` from a libmongoc error and a server reply to a
+/// `BulkWriteOperation`. If a partial result is provided, an updated result with the failed results filtered out will
+/// be returned as part of the error.
+internal func extractBulkWriteError(for: BulkWriteOperation,
+                                    error: bson_error_t,
+                                    reply: Document,
+                                    partialResult: BulkWriteResult? = nil) -> Error {
     let fallback = RuntimeError.internalError(message: "Got error from the server but couldn't parse it. " +
             "Message: \(toErrorString(error))")
 
     do {
         var bulkWriteErrors: [BulkWriteError] = []
-        if let writeErrors = serverReply["writeErrors"] as? [Document] {
+        if let writeErrors = reply["writeErrors"] as? [Document] {
             bulkWriteErrors = try writeErrors.map { try BSONDecoder().decode(BulkWriteError.self, from: $0) }
         }
 
         // Need to create new result that omits the ids that failed in insertedIds.
         var errResult: BulkWriteResult?
         if let result = partialResult {
-            let ordered = try bulkOp.opts?.getValue(for: "ordered") as? Bool ?? true
+            let ordered = try `for`.opts?.getValue(for: "ordered") as? Bool ?? true
```

Comment:
rather than doing this kinda funky thing to use a keyword as a variable name, could we just update the method signature so it says like `for op: BulkWriteOperation`?